experimental/devices/ads1x15/ads1x15.go

```diff
@@ -409,14 +410,23 @@ func (p *analogPin) Read() (analog.Reading, error) {
 }
 
 func (p *analogPin) ReadContinuous() <-chan analog.Reading {
-	p.Halt()
-	reading := make(chan analog.Reading)
+	// We need to lock if there are multiple Halt or ReadContinuous
+	// calls simultaneously.
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	// First release the current continuous reading if there is one
+	if p.stop != nil {
+		p.stop <- struct{}{}
+		close(p.stop)
```

Comment:
Technically speaking you don't need to close it. In fact it's probably preferable.